tests/Validator/FOPCommandFormatsValidator.php

```diff
@@ -57,13 +57,13 @@ class FOPCommandFormatsValidator
      *             `fop.console.domain.action`
      *             action can contain '.' or '_'
      */
-    private const SERVICE_REGEXP = '#^fop\.console\.(?<domain>[[:alpha:]\._]+)\.(?<action>[[:alpha:]\._]+)\.command$#X';
+    private const SERVICE_REGEXP = '#^fop\.console\.(?<domain>[[:alpha:]_]+)\.(?<action>[[:alpha:]_]+)\.command$#X';
```

Comment:
Don't you think it's confusing ?
Can you extract domain and action from `fop.console.customer.group.delete` ?
It could be customerGroup / delete or
Customer / group delete.
The less possibility we have, the less confusing it is.
Using dot for the main template and for word separation doesn't look a good idea to me.
Unless there is strong justification, is there any ?

Reply:
In my opinion there even should not be : in command name for separator.

Reply:
> Using dot for the main template and for word separation doesn't look a good idea to me.
Ah yea ok I see. Yes, same thing apply for command name then.
There is no strong justification, but it can allow better code organization. In a future version, we could imagine the possibility to allow subfolders in domains folders. So we would have for example:

Commands/
-- Image/
---- Generate/
![image](https://user-images.githubusercontent.com/48719560/143683823-581d5677-68f1-4544-ad93-9af932987ad0.png)

We just consider that the first folder in the domain, and we can add more for organizing the commands inside.
But maybe I go too far again... 😆 

> Can you extract domain and action from fop.console.customer.group.delete.command ?
> It could be customerGroup / delete or
> Customer / group delete.
The domain is always `customer` here, because dot is not allowed in domain name.

Finally, I suggest keeping same words separators for domain and action at the moment, so '-' for command name and '_' for service name. As you said, let's improve the code step by step according to our needs instead of planning the future. 😉

Reply:
> But maybe I go too far again...
yep, you are right, having subfolder will be a problem.
But that's not a problem we will face soon and maybe never.

Reply:
Currently, there is command names with ':' we'll have to rename them again :/ 
That's the price to pay.

Reply:
Ah yea, the images generation ones, with some aliases to keep BC.